### ISSUE TITLE
Web apps: remote debugging

### ISSUE THREAD

**Issue by maartenba** [JetBrains/azure-tools-for-intellij#85]
Will need Rider's remote debugger, then investigate feasibility.

**Comment by UnoSD** [comment]
Not sure if the implementation will take that into account, but would be nice to be able to debug also Azure Functions hosted on both Windows and Linux plans. This issue is quite old, can I please ask if there is any roadmap for this already?

**Comment by triforcely** [comment]
Is there any update regarding this issue?

**Comment by maartenba** [comment]
@triforcely unfortunately not at this moment

**Comment by triforcely** [comment]
@maartenba looks like good old logging every line will be needed 😅. Thanks for the response.

**Comment by mochsner** [comment]
Are there any plans to implement this feature still?

**Comment by megant** [comment]
It's a must have in 2021.

**Comment by rafek1241** [comment]
As you can see it's still not there.

**Comment by maartenba** [comment]
Correct. Haven't thought up reliable magic yet to be able to open up a port from within Azure. There is an out of the box experience that does this for Microsoft's debugger, but we are [not allowed to use the Microsoft debugger](https://visualstudio.microsoft.com/license-terms/mt644895/) unfortunately (article 1).

**Comment by furkankurt1** [comment]
I pay tens of hundreds of euros for this product. and have to open Visual studio to diagnose the issue. Shame! We need this. You have done a fantastic job. But need more.

**Comment by dpkblg** [comment]
2022, feature still not here. Disappointing :(

**Comment by DenisIvanovIvanov** [comment]
Still not here :)

**Comment by maartenba** [comment]
Nothing in Azure has changed to make this possible :-( https://github.com/JetBrains/azure-tools-for-intellij/issues/85#issuecomment-937782516

**Comment by triforcely** [comment]
@maartenba was there any effort from JetBrains to talk with Microsoft/Azure about this?

**Comment by maartenba** [comment]
Yes there has been.

On a side note, I just thought of a way this _could_  work, at least when the web app runs on Linux or on a container. The `az webapp create-remote-connection` command can be used to open a tunnel, and theoretically Rider's remote debugger can connect over it. Haven't tested, but may be a workaround for this.

**Comment by rgrace-puck** [comment]
> The `az webapp create-remote-connection` command can be used to open a tunnel, and theoretically Rider's remote debugger can connect over it.

Just tried this and the SSH connection succeeds, but the SFTP connection fails. Seems like it's required for the remote debug option to work.

**Comment by K3UL** [comment]
@maartenba 
Now that Visual Studio for Mac has been officially retired, do you think maybe this could be discussed with MS, at least regarding an exception for Rider on MacOS ?

Because I don't think even VsCode offers this for Asp.net code.

And even then, they have to be aware that no one doing actual C# dev wants to go from VS to VsCode with their half-baked extension or VS in a Windows VM (an actual alternative they suggest)

**Comment by rafaelldi** [comment]
https://github.com/JetBrains/azure-tools-for-intellij/pull/1001
